test/functional/outputs/azuremonitor/forward_to_azuremonitor_test.go

```diff
@@ -0,0 +1,194 @@
+// go:build !fluentd
+package azuremonitor
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"net/url"
+	"strings"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	logging "github.com/openshift/cluster-logging-operator/apis/logging/v1"
+	"github.com/openshift/cluster-logging-operator/internal/constants"
+	"github.com/openshift/cluster-logging-operator/internal/runtime"
+	"github.com/openshift/cluster-logging-operator/internal/utils"
+	"github.com/openshift/cluster-logging-operator/test/framework/functional"
+	"github.com/openshift/cluster-logging-operator/test/helpers/azuremonitor"
+	"github.com/openshift/cluster-logging-operator/test/helpers/oc"
+	"github.com/openshift/cluster-logging-operator/test/helpers/rand"
+	"github.com/openshift/cluster-logging-operator/test/helpers/types"
+	v1 "k8s.io/api/core/v1"
+)
+
+const (
+	azureSecret         = "azure-secret"
+	application         = "application"
+	serverStarted       = "Server started on port 3000"
+	transactionRecorded = "Transaction recorded"
+	failedReason        = "reason=\"Service call failed. No retries or retries exhausted.\""
+)
+
+var _ = Describe("Forwarding to Azure Monitor Log ", func() {
+	var (
+		framework  *functional.CollectorFunctionalFramework
+		mockServer *azuremonitor.MockoonServer
+		secret     *v1.Secret
+		sharedKey  = rand.Word(16)
+		inputName  = "custom-app"
+	)
+
+	BeforeEach(func() {
+		framework = functional.NewCollectorFunctionalFrameworkUsingCollector(logging.LogCollectionTypeVector)
+		mockServer = azuremonitor.NewMockoonServer(framework.Namespace, framework.Labels)
+
+		// Set up the common template forwarder configuration.
+		// HTTP Data Collector API URI	has format
+		// https://<CustomerId>.<Host>/api/logs?api-version=2016-04-01
+		// <CustomerID>	The unique identifier for the Log Analytics workspace.
+		// <Host> alternative host for dedicated Azure regions.
+		// We will split route host by dot (.) and use first part as customer id, second part as alternative host
+		// this allow as emulate forwarding
+		hostname, err := getServerHost()
+		Expect(err).To(BeNil())
+		subDomain := strings.SplitN(hostname, ".", 2)
+		framework.Forwarder.Spec.Outputs = append(framework.Forwarder.Spec.Outputs,
```

Comment:
nit. There are "builder" functions that may make this easier to read but not necessary